controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,255 @@
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	ocsVersion, err := getOcsVersion(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+	cephFSId, err := getCephFsid(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	odfDeploymentType := odfDeploymentTypeExternal
+	if !storageCluster.Spec.ExternalStorage.Enable {
+		odfDeploymentType = odfDeploymentTypeInternal
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(storageCluster); err != nil {
+		return "", err
+	}
+
+	connectedClients, err := getConnectedClients(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	data := OdfInfoData{
+		Version:           ocsVersion,
+		DeploymentType:    odfDeploymentType,
+		StorageSystemName: storageSystemName,
+		Clients:           connectedClients,
+		StorageCluster: infoStorageCluster{
+			NamespacedName:          client.ObjectKeyFromObject(storageCluster),
+			StorageProviderEndpoint: storageCluster.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return string(yamlData), nil
+
+}
+
+func getConnectedClients(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) ([]connectedClient, error) {
+	storageConsumers := &ocsv1a1.StorageConsumerList{}
+	err := r.Client.List(r.ctx, storageConsumers, client.InNamespace(storageCluster.Namespace))
+	if err != nil {
+		return nil, err
+	}
+	connectedClients := make([]connectedClient, 0, len(storageConsumers.Items))
+
+	for storageConsumerIdx := range storageConsumers.Items {
+		storageConsumer := &storageConsumers.Items[storageConsumerIdx]
+		clusterID := storageConsumer.Status.Client.ClusterID
+		name := storageConsumer.Status.Client.Name
+		newConnectedClient := connectedClient{
+			Name:      name,
+			ClusterID: clusterID,
+		}
+		connectedClients = append(connectedClients, newConnectedClient)
+	}
+
+	return connectedClients, nil
+}
+
+func getStorageSystemName(storageCluster *ocsv1.StorageCluster) (string, error) {
+	storageSystemRef := util.Find(storageCluster.OwnerReferences, func(ref *metav1.OwnerReference) bool {
+		return ref.Kind == "StorageSystem"
+	})
+	if storageSystemRef != nil {
+		return storageSystemRef.Name, nil
+	}
+
+	return "", fmt.Errorf(
+		"failed to find parent StorageSystem's name in StorageCluster %q"+
+			" ownerreferences, %v",
+		storageCluster.Name,
+		storageCluster.OwnerReferences)
+
+}
+
+func getOcsVersion(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	var csvs operatorsv1alpha1.ClusterServiceVersionList
+	err := r.Client.List(r.ctx, &csvs, client.InNamespace(storageCluster.Namespace))
```

Comment:
@raaizik shouldn't this be `operatorNamespace` instead of `storageCluster.Namespace` ??
In case of multiple StorageSystems/StorageClusters, each system/cluster will be in different namespace, but CSV will always be in operator installation namespace. Am I missing something ??

Reply:
I think you're right. I guess I didn't take multi SCs into account here.